src/Mbstring/bootstrap.php

```diff
@@ -42,88 +42,88 @@ function mb_list_encodings() { return p\Mbstring::mb_list_encodings(); }
+    function mb_strrichr($haystack, $needle, $before_needle = false, $encoding = null) { return p\Mbstring::mb_strrichr($haystack, $needle, $before_needle, $encoding); }
 }
 if (!function_exists('mb_strripos')) {
-    function mb_strripos($s, $needle, $offset = 0, $enc = null) { return p\Mbstring::mb_strripos($s, $needle, $offset, $enc); }
+    function mb_strripos($haystack, $needle, $offset = 0, $encoding = null) { return p\Mbstring::mb_strripos($haystack, $needle, $offset, $encoding); }
 }
 if (!function_exists('mb_strrpos')) {
-    function mb_strrpos($s, $needle, $offset = 0, $enc = null) { return p\Mbstring::mb_strrpos($s, $needle, $offset, $enc); }
+    function mb_strrpos($haystack, $needle, $offset = 0, $encoding = null) { return p\Mbstring::mb_strrpos($haystack, $needle, $offset, $encoding); }
 }
 if (!function_exists('mb_strstr')) {
-    function mb_strstr($s, $needle, $part = false, $enc = null) { return p\Mbstring::mb_strstr($s, $needle, $part, $enc); }
+    function mb_strstr($haystack, $needle, $before_needle = false, $encoding = null) { return p\Mbstring::mb_strstr($haystack, $needle, $before_needle, $encoding); }
 }
 if (!function_exists('mb_get_info')) {
     function mb_get_info($type = 'all') { return p\Mbstring::mb_get_info($type); }
 }
 if (!function_exists('mb_http_output')) {
-    function mb_http_output($enc = null) { return p\Mbstring::mb_http_output($enc); }
+    function mb_http_output($encoding = null) { return p\Mbstring::mb_http_output($encoding); }
 }
 if (!function_exists('mb_strwidth')) {
-    function mb_strwidth($s, $enc = null) { return p\Mbstring::mb_strwidth($s, $enc); }
+    function mb_strwidth($string, $encoding = null) { return p\Mbstring::mb_strwidth($string, $encoding); }
 }
 if (!function_exists('mb_substr_count')) {
-    function mb_substr_count($haystack, $needle, $enc = null) { return p\Mbstring::mb_substr_count($haystack, $needle, $enc); }
+    function mb_substr_count($haystack, $needle, $encoding = null) { return p\Mbstring::mb_substr_count($haystack, $needle, $encoding); }
 }
 if (!function_exists('mb_output_handler')) {
-    function mb_output_handler($contents, $status) { return p\Mbstring::mb_output_handler($contents, $status); }
+    function mb_output_handler($string, $status) { return p\Mbstring::mb_output_handler($string, $status); }
 }
 if (!function_exists('mb_http_input')) {
     function mb_http_input($type = '') { return p\Mbstring::mb_http_input($type); }
 }
 if (!function_exists('mb_convert_variables')) {
-    function mb_convert_variables($toEncoding, $fromEncoding, &$a = null, &$b = null, &$c = null, &$d = null, &$e = null, &$f = null) { return p\Mbstring::mb_convert_variables($toEncoding, $fromEncoding, $a, $b, $c, $d, $e, $f); }
+    function mb_convert_variables($to_encoding, $from_encoding, &$var, &...$vars) { return p\Mbstring::mb_convert_variables($to_encoding, $from_encoding, $var, ...$vars); }
```

Comment:
this is how PHP defines `mb_convert_variables`, but this is not supported in PHP 5.3 🤔 what should we do here?

Reply:
Conditional include

Reply:
should we remove it from `Mbstring::mb_convert_variables` and implement the function alone i seperate files ( e.g: `mb_convert_variables.5.php` / `mb_convert_variables.8.php` ) ?

Reply:
&... is supported since 5.6, so the new syntax is loaded in PHP >= 5.6, this allows passing more arguments, which matches the original implementation signature.

Reply:
hm [&$var, ...$vars] is only supported in >= 7.4 ( the only way to prepend an item while keeping the reference that I know of, when using `array_unshift($vars, $var)` we lose the reference to `$var` ), so the new signature is loaded only on >= 7.4